retry_test.go

```diff
@@ -204,3 +205,170 @@ func TestRoundTripConcurrencySafety(t *testing.T) {
 		assert.GreaterOrEqual(t, res.Delay, initialDelay, res.Res)
 	}
 }
+
+// TestDefaultRetryPolicy_IsErrorRetryable ensures that the IsErrorRetryable
+// method of DefaultRetryPolicy behaves as expected, correctly identifying
+// retryable and non-retryable errors.
+func TestDefaultRetryPolicy_IsErrorRetryable(t *testing.T) {
+	policy := DefaultRetryPolicy{}
+
+	err := (error)(nil)
+	if !policy.IsErrorRetryable(err) {
+		t.Errorf("IsErrorRetryable(nil) = false; want true")
+	}
+
+	err = errors.New("connection refused")
+	if !policy.IsErrorRetryable(err) {
+		t.Errorf("IsErrorRetryable(%v) = false; want true", err)
+	}
+
+	err = errors.New("unknown error")
+	if policy.IsErrorRetryable(err) {
+		t.Errorf("IsErrorRetryable(%v) = true; want false", err)
+	}
```

Comment:
For repeated test logic like this we try to use [Table Driven Tests](https://dave.cheney.net/2019/05/07/prefer-table-driven-tests).
See an example [here](https://github.com/mt-sre/client/blob/main/retry_test.go#L28-L96).
Makes it a lot easier to maintain as we just have to update the inputs and expected values to add new tests rather than copy/paste the test logic.